src/scripts/main.js

```diff
@@ -1,3 +1,135 @@
 'use strict';
 
-// write code here
+/**
+ * API Url:
+- Details URL:
+  https://mate-academy.github.io/phone-catalogue-static/api/phones/:phoneId.json
+
+Functions:
+- `getFirstReceivedDetails`
+  takes array of phone's ID and `resolves` with the first received detail
+  (the fastest response NOT the first in the list). Ignore the other responses;
+- `getAllSuccessfulDetails`
+  which takes array of phones' IDs and `resolves` with an array of all
+  successfully received details (errors should be ignored).
+
+- `getThreeFastestDetails`
+  which takes array of phones IDs and `resolves` with an array of the details
+  for the first 3 responses (the fastest).
+
+*/
+
+const BASE_URL = 'https://mate-academy.github.io/'
+  + 'phone-catalogue-static/api/phones';
+
+const request = (url) => {
+  return fetch(url)
+    .then((response) => {
+      // Create a promise that rejects in 5 milliseconds
+      return Promise.race([
+        new Promise((resolve, reject) => {
+          setTimeout(() => {
+            reject(new Error(`Error: ${response.status}`));
+          }, 5000);
+        }),
+        new Promise((resolve, reject) => {
+          if (response.ok) {
+            resolve(response.json());
+          } else {
+            reject(new Error('Can not connect to the server.'));
+          }
+        }),
+      ]);
+    });
+};
+
+const getPhoneIds = (startTime) => {
+  const timeout = 15000;
+
+  return request(`${BASE_URL}.json`)
+    .then(phones => phones.map(phone => phone.id))
+    .catch(err => {
+      if (startTime + timeout < Date.now()) {
+        return getPhoneIds();
+      }
+
+      return Promise.reject(err);
+    });
+};
+
+const getFirstReceivedDetails = (phoneIds) => {
+  return Promise.any(phoneIds.map(id => request(`${BASE_URL}/${id}.json`)));
+};
+
+const getAllSuccessfulDetails = (phoneIds) => {
+  return Promise.allSettled(
+    [...phoneIds].map(id => request(`${BASE_URL}/${id}.json`))
+  );
+};
```

Comment:
seems like those should be separated from main.js - they're responsible for different logic.